resources/documentation/stories.rst

```diff
@@ -0,0 +1,467 @@
+          c001:
+            bmc:                        # This instruction defines an attached BMC
+              name: bmanagement1        # This is the hostname of the BMC
+              ip4: 10.10.3.1            # This is the ipv4 of the BMC
+              mac: 08:00:27:dc:f8:f6    # This is the MAC hardware address of the BMC (for DHCP)
+              network: ice1-1           # This is the logical network this interface is connected to. Logical networks will be seen later.
+            network_interfaces:         # This is an instruction, to define bellow all host's NIC (Network Interface Controllers)
+              - interface: enp0s3       # This is the NIC name ('ip a' command to get NIC list)
+                ip4: 10.10.3.1          # This is the expected ipv4 for this NIC
+                mac: 08:00:27:dc:f8:f5  # This is the NIC MAC address, for the DHCP
+                network: ice1-1         # This is the logical network this NIC is linked to
+              - interface: ib0          # This is another NIC, not in the dhcp so no MAC is provided
+                ip4: 10.20.3.1
+                network: interconnect-1
+
+Then use the ansible-inventory command to check that the new host is listed on
+the configuration and seen by Ansible:
+
+.. code-block:: text
+
+  [root@management1 ~]# ansible-inventory --graph
+  [WARNING]: Unable to parse /etc/bluebanquise/internal as an inventory source
+  @all:
+    |--@mg_computes:
+    |  |--@equipment_typeC:
+    |  |  |--c001
+    |--@mg_logins:
+    |  |--@equipment_typeL:
+    |  |  |--login1
+    |--@mg_managements:
+    |  |--@equipment_typeM:
+    |  |  |--management1
+    |--@ungrouped:
+  [root@management1 ~]#
+
+Now, since we added a new node, replay the playbooks on management nodes (see
+the related story above) and if you are using the hosts_file role on all the
+cluster nodes, also replay their playbook, maybe limiting the execution to the
+needed roles, using tags. For example:
+
+.. code-block:: text
+
+  ansible-playbook /etc/bluebanquise/playbooks/computes.yml -t hosts_file
+
+Adding a new range of nodes
+---------------------------
+
+You may need to add a new range of nodes. You can do it manually, adding them
+one by one, or simply use a small bash script to generate the content you need.
+
+We assume here you need to generate a new range of c[1-4] of nodes, with ip
+on range 10.10.3.[1-4]. Same kind of ranges for their BMC and interconnect.
+
+Create a file /root/gen.sh with the following content:
+
+.. code-block:: bash
+
+  #!/bin/bash
+  cat <<EOF > computes.yml
+  mg_computes:
+    children:
+      equipment_typeC:
+        hosts:
+  EOF
+  for ((i=1;i<=$1;i++)); do
+  cat <<EOF >> computes.yml
+          c$i:
+            bmc:
+              name: bc$i
+              ip4: 10.10.103.$i
+              mac:
+              network: ice1-1
+            network_interfaces:
+              - interface: enp0s9
+                ip4: 10.10.3.$i
+                mac:
```

Comment:
Question from students: how can I define the MAC address afterwards?

Reply:
Manually :-|
Maybe I can provide a better script that accept an input file, with 2 columns, 1 for BMC MACs and 1 for nodes MACs ?

Reply:
I don't expect an answer or a solution. Just be ready to answer this question during any training/demo.  :grinning: